File: solutions/python/leap/1/leap.py
def leap_year(year):
    """
    Check if a given year is a leap year.,

    :param year: int - the year to check.
    :return: bool - True if the year is a leap year, False otherwise.
    """
    if year % 4 == 0:
        if year % 100 == 0:
            if year % 400 == 0:
                return True
            else:
                return False
        else:
            return True
    else:
        return False

File: solutions/c/leap/1/leap.c
#include "leap.h"

bool leap_year(int year) {
    return year % 4 == 0 && (year % 100 != 0 || year % 400 == 0);
}





#ifndef LEAP_H
#define LEAP_H

#include <stdbool.h>

bool leap_year(int year);

#endif